pytorch_lightning/accelerators/tpu.py

```diff
@@ -29,25 +25,6 @@
 class TPUAccelerator(Accelerator):
     """Accelerator for TPU devices."""
 
-    def setup(self, trainer: "pl.Trainer") -> None:
-        """
-        Raises:
-            ValueError:
-                If the precision or training type plugin are unsupported.
-        """
-        if not isinstance(self.training_type_plugin.precision_plugin, TPUPrecisionPlugin):
-            # this configuration should have been avoided in the accelerator connector
-            raise ValueError(
-                f"The `TPUAccelerator` can only be used with a `TPUPrecisionPlugin`,"
-                f" found: {self.training_type_plugin.precision_plugin}."
-            )
-        if not isinstance(self.training_type_plugin, (SingleTPUPlugin, TPUSpawnPlugin)):
-            raise ValueError(
-                "The `TPUAccelerator` can only be used with a `SingleTPUPlugin` or `TPUSpawnPlugin,"
-                f" found {self.training_type_plugin}."
-            )
-        return super().setup(trainer)
-
```

Comment:
reviewers: this has been moved to the end of the accelerator_connector, will be called at the end of the accelerator_connector init

Reply:
why not move the gpu/cpu acelerator checks to the accelerator connector as well?

Reply:
It's a bit different. This TPU check is checking what strategies and precision plugins to use when device is TPU, which should happens before in plugin selection phase like accelerator_connector.

For device specific checks like what's in cpu/gpu now, I think it will make sense to slim the accelerator_connector and moving to accelerator subclass.